### PR TITLE
API for password reset

### LINKED ISSUE #19: Need an API for password reset
**Is your feature request related to a problem? Please describe.**
API for a password reset, To allow the users to change their passwords in case they forget it.

**Describe the solution you'd like**
1. Get the email address from the user.
2. Send an email containing a unique reset link.
3. Redirect the user to the password reset page after validating the link.

### PR DESCRIPTION
**Pull request by RajeswarSharma**
[](https://pr-36.company.com) [PR 36](https://pr-36.company.com) RajeswarSharma Test plan  Completed [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=cu-coders&utm_campaign=badge_info)

fix #19 
**API URI** : https://main-cu-coders.herokuapp.com/forget
**Method**: POST
**Expected params**: email
**Responses :**

| property name | value | message                                        | description                                |
|---------------|-------|------------------------------------------------|--------------------------------------------|
| success       | true  | Please visit your e-mail inbox for reset link | Reset link generated and sent to the user |
| success       | false | User is not registered                         | Email id is not valid                      |
| success       | false | Internal server Error                          | Database connection error                  |

**Postman results**
image

### CODE REVIEW

File: configs/multer_config.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+const multer = require("multer");
+const path = require("path");
+const isImage = require("is-image");
+module.exports = multer({
+  storage: multer.diskStorage({
+    // extentions
+    filename: function (request, file, callback) {
+      const suff = Math.round(Math.random() * 1e9);
```
Comment (ank1traj) [inline configs/multer_config.js:8]: crypto.pseudoRandomBytes()/Math.random() is a cryptographically weak random number generator.

https://stackoverflow.com/questions/5651789/is-math-random-cryptographically-secure

**Comment by ank1traj** [comment]
https://deepsource.io/gh/cu-coders/backend/run/0d1b636a-0c64-4135-a2cb-aa72ef887b3b/javascript/

File: configs/multer_config.js (hunk 1 above)
Comment (RajeswarSharma) [reply configs/multer_config.js]: Removed line 8 as it is not required in cloudinary

Comment (RajeswarSharma) [reply configs/multer_config.js]: > https://deepsource.io/gh/cu-coders/backend/run/0d1b636a-0c64-4135-a2cb-aa72ef887b3b/javascript/

fixed the necessary ones.

**Pull request merged**